app/jobs/retrieve_trn_job.rb

```diff
@@ -15,8 +15,7 @@ def perform(trainee_id)
     trn = Dttp::RetrieveTrn.call(trainee: trainee)
 
     if trn
-      # TODO: make this a state transition
-      trainee.update!(trn: trn)
+      trainee.transition_to!(:trn_received) && trainee.update!(trn: trn)
```

Comment:
Since you'd always want to update the trn after transitioning to trn_received, perhaps this should be one method on a trainee 🤔

Reply:
Yes, I think that's a good shout.